### DISCUSSION TITLE
Solar inverter RCT Power

### DISCUSSION THREAD

**Discussion by chriskmn** [fhempy/fhempy#63] [Ideas]
Hi Dominik,

do you think it would be somehow possible to integrate a routine from the opensource project „OpenWB“ for reading out system variables from an RCT POWER solar inverter ?

There are only two major files, one a library, to get this done and FHEM doesn‘t need the ramdsik solution used by openwb:
[https://github.com/snaptec/openWB/blob/master/modules/bezug_rct2/rct_lib.py](url)
[https://github.com/snaptec/openWB/blob/master/modules/bezug_rct2/rct_read.py](url)

But it would be nice if it could be defined via FHEM wich parameters should be read. The inverter has a ton of internal values. 
This is an example how an other fellow integrated it, but unfortunately with performance issues:
[https://github.com/marvin78/FHEM-RCT/blob/main/FHEM/37_RCT.pm](url)

There is even a simulator to test the code:
[https://rctclient.readthedocs.io/en/latest/simulator.html](url)

I would be happy to help if I could get a little direction how to start. I am only a mechanical engineer 🙈

Cheers, Chris

**Comment by fhempy** [comment]
Hi Chris,

I had a look at the rctclient and it looks possible to integrate it. Can you please let me know how you use the RCT module in FHEM? Do you just specify the IP address to connect to or do you provide further parameters as well?
